### PR TITLE
Integrating Prometheus with /metrics endpoint (v2)

### PR DESCRIPTION
**Pull request by punkupoz**
Another implementation of [PR 13](https://github.com/andytruong/redes-writer/pull/13).

This PR does not change function signature of Run, loop, ...
However, It does not support configuration of which metric to collect.

### CODE REVIEW

**Comment by punkupoz** [comment]
@andytruong

File: docker-compose.yml [hunk 1]
```diff
@@ -26,3 +26,21 @@ services:
...
+      test: ["CMD-SHELL", "wget localhost:8484/metrics -q -O - > /dev/null 2>&1"]
+      interval: 30s
+      timeout: 30s
+      retries: 3
+  prometheus:
+    image: "prom/prometheus:v2.12.0"
+    ports:
+      - "9090:9090"
+    volumes:
+      - "./metrics.configs.sample/prometheus/prometheus.sample.yml://etc/prometheus/prometheus.yml"
```
Comment (andytruong) [inline docker-compose.yml]: need better path than `metrics.configs.sample/prometheus/prometheus.sample.yml`.

File: queue.go [hunk 2]
```diff
@@ -68,7 +68,19 @@ func (q queue) Listen(ctx context.Context, errCh chan error) chan string {
 }
 
 func (q *queue) loop(ctx context.Context, sub chan string, ch chan string) {
+	pt := newHistogram("process_time", "Process time of each loop", []float64{0.025, 0.05, 0.075, 0.1, 0.125}).register()
```
Comment (andytruong) [inline queue.go]: discussed on slack, should put in NewProcessor().

File: metrics.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+type (
+	histogram struct {
+		prometheus.Histogram
+	}
+)
+
+// Return a metric collector that ready to be registered into /metrics endpoint
+func newHistogram(name string, help string, buckets []float64) *histogram {
```
Comment (andytruong) [inline metrics.go]: should have a better name, processSpeedHistogram?

Comment (punkupoz) [reply metrics.go]: This method will create a new histogram with name passed into the "name" parameter, not specifically for processSpeed histogram.

File: docker-compose.yml (hunk 1 above)
Comment (punkupoz) [reply docker-compose.yml]: This hierachy is necessary. "metrics.configs.sample/" will contains configs of alert manager, grafana,... Prometheus is going to have "alert.rules". Having the hierachy will make it easier to create, find and use configs later on.

Comment (punkupoz) [reply docker-compose.yml]: We can change the name of "metrics.configs.sample" directory.

**Pull request merged**